source/base/process_grid.cc

```diff
@@ -73,7 +73,7 @@ namespace
     // 2x5.
 
     // limit our estimate to be in [2, Np]
-    int n_process_columns = std::min(Np, std::max(2, Pc));
+    int n_process_columns = std::clamp(Pc, 2, Np);
```

Comment:
I think you have the order wrong here, in part because the comment is wrong: The original code is equivalent to
```suggestion
    int n_process_columns = std::clamp(Pc, Np, 2);
```
This triggers an abort in the tests at some point whereas the old code worked. (I have no idea what the code actually does, and will admit that it looks funny as is.)

In fact, on second thought, I think one of the issues is that it's not clear to me what is supposed to happen if `Np==1`. I think that is the case that aborts. My suggestion is to just undo this one change and see that we can at least get the rest merged.

Reply:
I tried reading through proccess_grid.cc and have reverted the changes. Thank you guys for the help!